pkg/controller/core.oam.dev/v1alpha2/applicationcontext/applicationcontext_controller.go

```diff
@@ -83,7 +82,9 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	}
 	appConfig.Status = appContext.Status
 	// the name of the appConfig has to be the same as the appContext
-	appConfig.ObjectMeta = metav1.ObjectMeta{Namespace: appContext.Namespace, Name: appContext.Name, UID: appContext.UID}
```

Comment:
AC controller needs ownerRef to determine isControlledByApp. So we should not override ownerRef here.